libclc/CMakeLists.txt

```diff
@@ -1,47 +1,110 @@
 cmake_minimum_required(VERSION 3.20.0)
 
-add_custom_target(libspirv-builtins COMMENT "Build libspirv builtins")
-add_custom_target(libclc-builtins COMMENT "Build libclc builtins")
-
-# Add path for custom modules
-set(CMAKE_MODULE_PATH
-  ${CMAKE_CURRENT_SOURCE_DIR}/cmake
-  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules
-  ${CMAKE_MODULE_PATH})
+project( libclc VERSION 0.2.0 LANGUAGES CXX C)
 
-# If we are not building as a part of LLVM, build libclc as an
-# standalone project, using LLVM/Clang as external tools.
-if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR )
-  project( libclc VERSION 0.2.0 LANGUAGES CXX C )
+set(CMAKE_CXX_STANDARD 17)
 
-  set( LIBCLC_STANDALONE_BUILD 1 )
+# Add path for custom modules
+list( INSERT CMAKE_MODULE_PATH 0 "${PROJECT_SOURCE_DIR}/cmake/modules" )
 
-  include(HandleOutOfTreeLLVM)
-else()
-  include(HandleInLLVMTree)
-endif()
+set( LIBCLC_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR} )
+set( LIBCLC_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR} )
+set( LIBCLC_OBJFILE_DIR ${LIBCLC_BINARY_DIR}/obj.libclc.dir )
 
-set(CMAKE_CXX_STANDARD 17)
+include( AddLibclc )
 
 include( GNUInstallDirs )
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS
   amdgcn-amdhsa/lib/SOURCES;
+  amdgcn-amdhsa/libspirv/SOURCES;
   amdgcn/lib/SOURCES;
+  amdgcn/libspirv/SOURCES;
   amdgcn-mesa3d/lib/SOURCES;
   amdgpu/lib/SOURCES;
   clspv/lib/SOURCES;
   clspv64/lib/SOURCES;
   generic/lib/SOURCES;
+  generic/libspirv/SOURCES;
   ptx/lib/SOURCES;
   ptx-nvidiacl/lib/SOURCES;
+  ptx-nvidiacl/libspirv/SOURCES;
   r600/lib/SOURCES;
+  r600/libspirv/SOURCES;
   spirv/lib/SOURCES;
   spirv64/lib/SOURCES
+  x86_64-unknown-linux/libspirv/SOURCES
 )
 
-include( AddLibclc )
+set( LIBCLC_MIN_LLVM 3.9.0 )
+
+set( LIBCLC_TARGETS_TO_BUILD "all"
+    CACHE STRING "Semicolon-separated list of targets to build, or 'all'." )
```

Comment:
Might want to mention libclc in the help message, because that wording matches LLVM_TARGETS_TO_BUILD and LLVM_ENABLE_PROJECTS somewhat, and that might be a bit confusing on the command line

Reply:
Yeah that's probably a good idea. I haven't changed it in this PR, though. I think I'll open a PR with upstream and that'll filter back down to us. Does that sound okay?

Reply:
ack